frontend/src/__tests__/unittest/signup/Signup.test.tsx

```diff
@@ -0,0 +1,65 @@
+import { waitFor } from '@testing-library/react';
+import { vi } from 'vitest';
+import userEvent from '@testing-library/user-event';
+import { formSetup } from './Client-util';
+import {
+  INVALID_CONFIRMPWD_ERRORMESSAGE,
+  INVALID_EMAIL_ERRORMESSAGE,
+  INVALID_PWD_ERRORMESSAGE,
+  INVALID_USERNAME_ERRORMESSAGE,
+} from '@/constants/Stringconstant';
+const mockedUseNavigate = vi.fn();
+vi.mock('react-router-dom', async () => {
+  const mod = await vi.importActual<typeof import('react-router-dom')>('react-router-dom');
+  return {
+    ...mod,
+    useNavigate: () => mockedUseNavigate,
+  };
+});
+
+describe('form', async () => {
+  test('should display invalid error message when providing invalid inputs during signup', async () => {
+    const userActions = userEvent.setup();
+    const {
+      form,
+      usernameInput,
+      emailInput,
+      passwordInput,
+      confirmpasswordInput,
+      signupbuttonText,
+    } = await formSetup();
+    await userActions.type(usernameInput, 'abcd');
+    await userActions.type(emailInput, 'abc@');
+    await userActions.type(passwordInput, '123');
+    await userActions.type(confirmpasswordInput, '1234');
+    await userActions.click(signupbuttonText);
+    await waitFor(() => {
+      expect(form.getByText(INVALID_USERNAME_ERRORMESSAGE));
+      expect(form.getByText(INVALID_EMAIL_ERRORMESSAGE));
+      expect(form.getByText(INVALID_PWD_ERRORMESSAGE));
+      expect(form.getByText(INVALID_CONFIRMPWD_ERRORMESSAGE));
+    });
```

Comment:
Can we have few more tests with just user_name wrong or password wrong and check if it is working, all permutations and combinations is even fine but additional 3 -4 will do